packages/react-dom-bindings/src/client/ReactFiberConfigDOM.js

```diff
@@ -2272,6 +2365,19 @@ function animateGesture(
     // keyframe. Otherwise it applies to every keyframe.
     moveOldFrameIntoViewport(keyframes[0]);
   }
+  if (unchangedDimensions && width !== undefined && height !== undefined) {
+    // Read the underlying width/height of the pseudo-element. The previous animation
+    // should have already been cancelled so we should observe the underlying element.
+    const computedStyle = getComputedStyle(targetElement, pseudoElement);
```

Comment:
I believe this technique doesn't work in Safari (and in fact since I used it all over the place, a ton of things are already broken). There's a bug where Safari doesn't return styles computed from View Transition pseudo elements and since Safari a piece of shit software I don't expect it to get fixed.